### PR TITLE
Tests: Add missing tests to mentorship relation

### LINKED ISSUE #990: Tests: Add missing tests in Mentorship Relation
## Description
Some tests are missing in `app/api/dao/mentorship_relation.py` file. Cover those to ensure any new change will not break existing changes.

## Test Cases
The test cases that should be cover are:-
- Same id for relation
https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L50

- Incorrect End Time
https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L59

- Incorrect Mentor time
https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L65

https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L68

- Mentor not available
https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L77

- Mentee not available
https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L86

- Uninvolve relation request
https://github.com/anitab-org/mentorship-backend/blob/2917a910ab34dd5b6f800e4eac08e8fa5ba3fa0f/app/api/dao/mentorship_relation.py#L265-L268

## Definition of Done
- [ ] At least one test case tested.
- [ ] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by Sanika1999**
### Description

Added Test case for MENTOR_ID_SAME_AS_MENTEE_ID

Fixes #990

### Type of Change:

-Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [X] Any dependent changes have been merged

**Code/Quality Assurance Only**

- [] My changes generate no new warnings 
- [X] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,4 +9,4 @@ Flask-Testing==0.8.0
 gunicorn==20.0.4
 psycopg2-binary==2.8.6
 python-dotenv==0.14.0
-six==1.11.0
+six==1.14.0
```
Comment (devkapilbansal) [inline requirements.txt:12]: Don't upgrade it as it may cause some dependency issues

File: tests/mentorship_relation/test_dao_creation.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from datetime import datetime, timedelta
 from http import HTTPStatus
 
+
```
Comment (devkapilbansal) [inline tests/mentorship_relation/test_dao_creation.py:5]: Why this extra line here?

File: tests/mentorship_relation/test_dao_creation.py [hunk 3]
```diff
@@ -249,6 +250,40 @@ def test_dao_create_mentorship_relation_with_good_args_but_invalid_timestamp(sel
...
+        self.assertIsNone(query_mentorship_relation)
+    # def test_dao_create_mentorship_relation_with_end_time_before_present(self):
+        
+    #     dao = MentorshipRelationDAO()
+    #     data = dict(
+    #         mentor_id=self.first_user.id,
+    #         mentee_id=self.second_user.id,
+    #         now_datetime = datetime.now(),
+    #         end_date = datetime.fromtimestamp(data["end_date"]),
```
Comment (devkapilbansal) [inline tests/mentorship_relation/test_dao_creation.py:276]: From where you are taking `data[end_date]`. data is not yet created and you are trying to access it. :monocle_face:

File: tests/mentorship_relation/test_dao_creation.py [hunk 4]
```diff
@@ -249,6 +250,40 @@ def test_dao_create_mentorship_relation_with_good_args_but_invalid_timestamp(sel
...
+    #         notes=self.notes_example,
+    #         tasks_list=TasksListModel(),
+    #     )
+    #     result = dao.create_mentorship_relation(self.first_user.id, data)
+    #     self.assertEqual(messages.END_TIME_BEFORE_PRESENT, result[0])
+    #     self.assertEqual(400, result[1])
+    
+
+       
+
```
Comment (devkapilbansal) [inline tests/mentorship_relation/test_dao_creation.py:286]: Consider removing empty lines when this PR is ready for a review

**Review by devkapilbansal — CHANGES_REQUESTED** [review]

**Comment by devkapilbansal** [comment]
@Sanika1999 any progress?

**Comment by Sanika1999** [comment]
> @Sanika1999 any progress?

yes iam still working on this, got a bit delayed because of exams

**Comment by isabelcosta** [comment]
@Sanika1999 are you still interested in working on this?
It's okay, if not. We can make the issue available again

**Comment by devkapilbansal** [comment]
Closing due to inactivity

**Pull request closed without merging**